core/modules/views/plugins/views_plugin_display_block.inc

```diff
@@ -134,7 +134,7 @@ class views_plugin_display_block extends views_plugin_display {
         $form['#title'] .= t('Block admin description');
         $form['block_description'] = array(
           '#type' => 'textfield',
-          '#description' => t('This will appear as the name of this block in administer >> structure >> blocks.'),
+          '#description' => t('This will appear as the name name of this block when adding it to a layout at administer >> structure >> layouts.'),
```

Comment:
"name name"?

It also probably wouldn't hurt to eliminate the path and just say "when adding it to a layout."